//===-- get_details.proto - request & response payloads -------------------===//
//
// Copyright (c) 2025 Nuclearist <nuclearist@teknology-hub.com>
// Part of tek-steamclient, under the GNU General Public License v3.0 or later
// See https://github.com/teknology-hub/tek-steamclient/blob/main/COPYING for
//    license information.
// SPDX-License-Identifier: GPL-3.0-or-later
//
//===----------------------------------------------------------------------===//
syntax = "proto3";

package tek.steamclient.cm.msg_payloads;

import "tek/steamclient/cm/msg_payloads/published_file_details.proto";

option optimize_for = LITE_RUNTIME;

// Request message payload for "PublishedFile.GetDetails#1".
message GetDetailsRequest {
  // IDs of the items to request details for.
  repeated fixed64 ids = 1;
  // Value indicating whether child item IDs should be provided for collections.
  optional bool include_children = 4;
}

// Response message payload for "PublishedFile.GetDetails#1".
message GetDetailsResponse {
  // Details for the requested items.
  repeated tek.steamclient.cm.msg_payloads.PublishedFileDetails details = 1;
}
